### PR TITLE
[#29803] Fix nil pointer access in logRuntimeDependencies

### LINKED ISSUE #29803: [Failing Test]: Python Container Builds Fail with SIGSEGV: segmentation violation
### What happened?

Python container builds that are invoked through `processArtifactsInSetupOnlyMode()` fail because of the newly introduced `logRuntimeDependencies()` function. It appears that the function always tries to invoke the logger directly instead of using a buffered logger, accessing a nil value. This should be fixed by accepting a buffered logger type instead, which accounts for this issue.  

```
goroutine 1 [running]:
github.com/apache/beam/sdks/v2/go/container/tools.(*Logger).Log(0x0, {0xa4c3a8, 0xe36cc0}, 0x2, {0xc000038640, 0x1d})
	/runner/_work/beam/beam/sdks/go/container/tools/logging.go:59 +0x66
github.com/apache/beam/sdks/v2/go/container/tools.(*Logger).Printf(0x0?, {0xa4c3a8, 0xe36cc0}, {0x99939e?, 0xc9f8ade50?}, {0x0?, 0x7fffffffffffffff?, 0xa4c3a8?})
	/runner/_work/beam/beam/sdks/go/container/tools/logging.go:106 +0x5a
main.logRuntimeDependencies({0xa4c3a8, 0xe36cc0}, 0x0)
	/runner/_work/beam/beam/sdks/python/container/boot.go:463 +0x5b
main.installSetupPackages({0xa4c3a8, 0xe36cc0}, 0x0, {0xc000042e30, 0x1, 0x1}, {0x7fff05c69db0, 0x1a}, {0xc00018fd80, 0x1, ...})
	/runner/_work/beam/beam/sdks/python/container/boot.go:412 +0x50b
main.processArtifactsInSetupOnlyMode()
	/runner/_work/beam/beam/sdks/python/container/boot.go:455 +0x51f
main.main()
	/runner/_work/beam/beam/sdks/python/container/boot.go:82 +0x76
```

### Issue Failure

Failure: Test is continually failing

### Issue Priority

Priority: 0 (outage / failing precommit test impacting development)

### Issue Components

- [X] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam YAML
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

### PR DESCRIPTION
**Pull request by jrmccluskey**
Corrects a nil pointer usage when Python containers are built through `processArtifactsInSetupOnlyMode()`

Fixes #29803

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by jrmccluskey** [comment]
R: @damccorm @riteshghorse 
CC: @tvalentyn

**Review by lostluck — APPROVED** [review]

**Review by riteshghorse — APPROVED** [review]

File: sdks/python/container/boot.go [hunk 1]
```diff
@@ -409,7 +409,7 @@ func installSetupPackages(ctx context.Context, logger *tools.Logger, files []str
 	if err := pipInstallPackage(ctx, logger, files, workDir, workflowFile, false, true, nil); err != nil {
```
Comment (lostluck) [inline sdks/python/container/boot.go:409]: Do we want to do the same for installExtraPackages and pipInstallPackages?

**Comment by lostluck** [comment]
It is an odd pattern that at present: 
* setup passes down a nil logger,
* then we manually re-build/wrap with buffered loggers wherever we actually do log something.

I'd expect us to pass around the buffered logger type from the top, instead of arbitrarily trying to re-construct it.

Stuff to clean up when it's not a release blocker.

File: sdks/python/container/boot.go (hunk 1 above)
Comment (jrmccluskey) [reply sdks/python/container/boot.go:409]: We can clean this up after the fact to a cleaner pattern, IIRC an issue similar to the one this fixes popped up when I was implementing the buffered logger originally which led to needing to handle the nil logger in the setup-only mode case.

Comment (tvalentyn) [reply sdks/python/container/boot.go:409]: Thanks, @jrmccluskey ! Could you please  file an issue to follow up? thank you!

**Pull request merged**